dbms/tests/queries/bugs/00597_push_down_predicate.sql

```diff
@@ -0,0 +1,18 @@
+DROP TABLE IF EXISTS test.test;
+DROP TABLE IF EXISTS test.test_view;
+
+CREATE TABLE test.test(date Date, id Int8, name String, value Int64) ENGINE = MergeTree(date, (id, date), 8192);
+CREATE VIEW test.test_view AS SELECT * FROM test.test;
+
+SET enable_optimize_predicate_expression = 1;
+SET enable_debug_queries = 1;
+
+-- Optimize predicate expression with view
+-- TODO: simple view is not replaced with subquery inside syntax analyzer
+ANALYZE SELECT * FROM test.test_view WHERE id = 1;
```

Comment:
Maybe the same as https://github.com/yandex/ClickHouse/pull/4387/files#r258121391 ?

Reply:
No, it's not the same - at least from the `SyntaxAnalyzer` point of view.